### ISSUE TITLE
WS+ Undo feature jeopardizes WSFW/PGT/TS power grid corruption detection

### ISSUE THREAD

**Issue by niston** [kinggath/Workshop-Plus#40]
Because the undo feature holds deleted items persistent, Delete() of a workshop-built powered object will not immediately corrupt the power grid. As a consequence, grid corruption detection by WSFW, TS or other capable mods is rendered unreliable: Only after the item drops out of the undo queue will the bug actually occur. The undo queue holds up to 1280 items, which means that the bug essentially gets postponed until a random (1...1279) number of new items has been built by the user, or the undo queue gets flushed for some reason.

I can easily imagine a situation resulting as such:
Mod A does a boo boo, user builds happily away because they trust WSFW to detect the issue and no warning is given. 2 Months later, user installs Mod B, builds a few items and *bam* the bug triggers as the long deleted item falls out of the queue. Mod B gets denounced as crap, trash, etc, whereas the real culprit Mod A will be very hard to identify.

Proposed fix is to completely exclude powered items from the undo history:
```
Function CreateHistory(ObjectReference akObjectRef, Int aiWorkshopID, Int aiActionType)
	If (akObjectRef.HasKeyword(PowerConnection) || akObjectRef.GetActorValue(SnapTransmitsPower) != 0.0)
		; do not store history for powered items (will jeopardize power grid corruption detection)
		return
	EndIf
	...
```
I would additionally recommend a MCM toggle to disable the Undo feature entirely, as I personally found the forced persistence to be often much more of a nuisance than it is helpful.

**Comment by niston** [comment]
Addendum: When an object is picked up in workshop mode, it gets added to the undo queue. This is true even for script spawned objects. Something to keep in mind when testing script spawned powered stuff!

**Comment by kinggath** [comment]
For next build, added a setting to disable Undo/Redo, or just exclude powered items. The new setting will do checks at CreateHistory moment to prevent inappropriate items from ending up in the queue. Undo is going to start disabled to avoid the unexpected persistence.
